src/main/scala/akka/rabbitmq/ConnectionActor.scala

```diff
@@ -91,10 +92,6 @@ class ConnectionActor(
 
     case Event(msg @ AmqpShutdownSignal(cause), Connected(connection)) =>
       // It is important that we check if a shutdown signal pertains to the current connection.
-      // This actor explicitly close the connection ourselves before reconnecting. So if this actor starts to reconnect
```

Comment:
Because I think the argument is more like: rabbitmq sends out a shutdown signal for any connection closing. Therefore we should check that the event we get here applies to the connection we opened. The first line of this comment captures this already.